src/main/java/org/humancellatlas/ingest/project/ProjectService.java

```diff
@@ -61,11 +64,31 @@ protected Logger getLog() {
     }
 
     public Project register(final Project project) {
+        project.setCataloguedDate(null);
+        if (!isNull(project.getIsInCatalogue()) && project.getIsInCatalogue()) {
+            project.setCataloguedDate(Instant.now());
+        }
         Project persistentProject = projectRepository.save(project);
         projectEventHandler.registeredProject(persistentProject);
         return persistentProject;
     }
 
+    public Project update(final Project project, ObjectNode patch, Boolean sendNotification) {
+        if (patch.has("isInCatalogue")
+            && patch.get("isInCatalogue").asBoolean()
+            && project.getCataloguedDate() == null){
+            project.setCataloguedDate(Instant.now());
```

Comment:
If the checkbox was checked after being unchecked, the catalogue date will always contain the date when it was first checked. Is that what we want? 

If not, maybe we could only update the catalogue date when isInCatalogue is false and being set to true?

Reply:
I think this is probably an edge case. Its could that we can remove projects from the catalogue if the box is unchecked, but in the scenario that it is "in the catalogue" then "out of the catalogue" then "in the catalogue" again, I think its ok to use the original "in" date. I hope that makes sense? So I think its fine as is...

Reply:
I see, thanks!

Reply:
great thinking on this Alegria! Its great how many scenarios you are thinking through here